### PR TITLE
Migrate python-versions pipeline from AzDO to GH Actions

### PR DESCRIPTION
**Pull request by mikhailkoliada**
As was requested, lets move the python package generation pipeline from AzDO to GH Actions. The only disadvantage is that we can not choose the specific OS we want to build/package a python version for, otherwise the functionality is pretty similar to what we are using now (we can specify a python version to build and decide if we need to publish a release or not).

Apart from that, I've removed a NUnit xml report generation in the tests block (AzDO has a nice feature for that to report the statistics, while on GH we have no proper action to publish the results of testing as there is no an appropriate official task to do that), as an alternative I propose we are using the `-PassThru` parameter on the `Invoke-Pester` invocation and fail the job based on the failed tests counter.

Also GH Actions does not have 1:1 correlation between pre-defined variables, that is why a separate directory for the artifact was created from within the builder script.

 **Original issue:**  https://github.com/actions/virtual-environments-internal/issues/3104

### CODE REVIEW

File: builders/python-builder.psm1 [hunk 1]
```diff
@@ -41,10 +41,13 @@ class PythonBuilder {
     PythonBuilder ([semver] $version, [string] $architecture, [string] $platform) {
         $this.InstallationTemplatesLocation = Join-Path -Path $PSScriptRoot -ChildPath "../installers"
 
-        $this.HostedToolcacheLocation = $env:AGENT_TOOLSDIRECTORY
-        $this.TempFolderLocation = $env:BUILD_SOURCESDIRECTORY
-        $this.WorkFolderLocation = $env:BUILD_BINARIESDIRECTORY
-        $this.ArtifactFolderLocation = $env:BUILD_STAGINGDIRECTORY
+        New-Item -Force -Type Directory (Join-Path $env:RUNNER_TEMP "artifact")
```
Comment (al-cheb) [inline builders/python-builder.psm1:44]: Create an additional variable for (Join-Path $env:RUNNER_TEMP "artifact") or move below $this.ArtifactFolderLocation and pass those variables.

File: .github/workflows/python-builder.yml [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+
+      - name: Verbose python binary links
+        run: |
+          $pythonLocation = which python
+          if ('${{ matrix.platform }}' -ne 'win32') {
+            if ('${{ matrix.platform }}' -eq 'darwin') { otool -L $pythonLocation } else { ldd $pythonLocation }
+          }
+
+      - name: Run tests
+        shell: pwsh
```
Comment (al-cheb) [inline .github/workflows/python-builder.yml:148]: nitpick: remove shell: pwsh

File: .github/workflows/python-builder.yml [hunk 3]
```diff
@@ -0,0 +1,220 @@
+name: Build python package
+
+on:
+  workflow_dispatch:
+    inputs:
+      VERSION:
+        description: 'Python version to build and upload'
+        default: '3.9.6'
```
Comment (miketimofeev) [inline .github/workflows/python-builder.yml:8]: time to update the default? 😉
```suggestion
        default: '3.9.9'
```

File: .github/workflows/python-builder.yml [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+          }
+        working-directory: ${{ runner.temp }}/${{ env.ARTIFACT_NAME }}
+
+      - name: Apply build artifact to the local machine
+        run: |
+          if ('${{ matrix.platform }}' -eq 'win32') { powershell ./setup.ps1 } else { sh ./setup.sh }
+          if ('${{ matrix.platform }}' -ne 'win32') {
+            cp ${{ runner.temp }}/${{ env.ARTIFACT_NAME }}/build_output.txt ${{ runner.temp }}
+          }
+
```
Comment (al-cheb) [inline .github/workflows/python-builder.yml:127]: remove a new line

**Review by miketimofeev — APPROVED** [review]

File: builders/python-builder.psm1 [hunk 5]
```diff
@@ -41,10 +41,16 @@ class PythonBuilder {
...
-        $this.ArtifactFolderLocation = $env:BUILD_STAGINGDIRECTORY
+        $artifactDirectory = Join-Path $env:RUNNER_TEMP "artifact"
+        $workDirectory = Join-Path $env:RUNNER_TEMP "work"
+
+        New-Item -Force -Type Directory $artifactDirectory
+        New-Item -Force -Type Directory $workDirectory
+
+        $this.HostedToolcacheLocation = $env:RUNNER_TOOL_CACHE
+        $this.TempFolderLocation = $env:RUNNER_TEMP
+        $this.WorkFolderLocation = Join-Path $env:RUNNER_TEMP "work"
```
Comment (miketimofeev) [inline builders/python-builder.psm1:52]: Don't we have a variable for this now?

**Review by miketimofeev — APPROVED** [review]

**Review by al-cheb — APPROVED** [review]

**Review by MaksimZhukov — APPROVED** [review]

**Pull request merged**